### PR TITLE
Add gnulib/20200224 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **gnulib/20200224**

Tested on Windows with MSVC

I think this might be useful for m4: #346 

CC'ing @SSE4

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/gnulib/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import shutil
+
+
+class GnuLibConanFile(ConanFile):
+    name = "gnulib"
+    description = "Gnulib is a central location for common GNU code, intended to be shared among GNU packages."
+    homepage =  "https://www.gnu.org/software/gnulib/"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "gnulib", "library", "gnu")
+    license = "GPL-3.0-or-later", "LGPL-3.0-or-later", "Public domain"
```
Comment (Croydon) [inline recipes/gnulib/all/conanfile.py]:
```suggestion
    license = ("GPL-3.0-or-later", "LGPL-3.0-or-later", "Public domain")
```

File: recipes/gnulib/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    def package(self):
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
+
+        # The following line did not work, so do it the long way...
+        # shutil.copy(os.path.join(self.source_folder, self._source_subfolder), os.path.join(self.package_folder, "bin"))
+
+        gnulib_dir = os.path.join(self.source_folder, self._source_subfolder)
+        for root, _, files in os.walk(gnulib_dir):
+            relpath = os.path.relpath(root, os.path.join(self.source_folder, self._source_subfolder))
```
Comment (madebr) [inline recipes/gnulib/all/conanfile.py]:
```suggestion
            relpath = os.path.relpath(root, gnulib_dir)
```

File: recipes/gnulib/all/test_package/test_package.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
+#include <ctype.h>
```
Comment (madebr) [inline recipes/gnulib/all/test_package/test_package.c]:
```suggestion
#include <config.h>

#include <ctype.h>
```

File: recipes/gnulib/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires("msys2/20190524")
+        self.build_requires("automake/1.16.1")
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                env = {
+                    "AR": "{} lib".format(os.path.join(self.build_folder, "build-aux", "ar-lib").replace("\\", "/")),
```
Comment (madebr) [inline recipes/gnulib/all/test_package/conanfile.py]:
```suggestion
                    "AR": "{} lib".format(tools.unix_path(os.path.join(self.build_folder, "build-aux", "ar-lib"))),
```

File: recipes/gnulib/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import shutil
+
+
+class GnuLibConanFile(ConanFile):
+    name = "gnulib"
+    description = "Gnulib is a central location for common GNU code, intended to be shared among GNU packages."
+    homepage =  "https://www.gnu.org/software/gnulib/"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "gnulib", "library", "gnu")
+    license = ("GPL-3.0-or-later", "LGPL-3.0-or-later", "Public domain")
```
Comment (uilianries) [inline recipes/gnulib/all/conanfile.py]:
```suggestion
    license = ("GPL-3.0-or-later", "LGPL-3.0-or-later", "Unlicense")
```

The unlicense is a public domain license: https://spdx.org/licenses/Unlicense.html

**Pull request merged**